qemu/tests/qemu_guest_agent.py

```diff
@@ -722,6 +722,40 @@ def _action_after_fsfreeze(self, *args):
                                  " write file")
 
     @error.context_aware
+    def gagent_check_reset_reboot_shutdown(self, test, params, env):
+
+        """
+        Execute "system_reset" and send "shutdown,reboot" command to guest agent after FS fsreezed
+        :param test: kvm test objecti
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+        """
+        vm = env.get_vm(params["main_vm"])
+        if not self.vm.is_alive():
+            test.skip("VM died,can't continue the test loop.")
+        self.gagent.fsfreeze()
+        self.gagent.verify_fsfreeze_status(self.gagent.FSFREEZE_STATUS_FROZEN)
+
+        try:
+            self.gagent.shutdown(self.gagent.SHUTDOWN_MODE_POWERDOWN)
+        except Exception:
+            self.test.fail("guest-shutdown command returned a wrong message")
+        else:
+            logging.info("guest FS is still freeze status")
+        self.gagent.verify_fsfreeze_status(self.gagent.FSFREEZE_STATUS_FROZEN)
+
+        try:
+            self.gagent.shutdown(self.gagent.SHUTDOWN_MODE_REBOOT)
+        except Exception:
+            self.test.fail("guest-reboot command returned a wrong message")
+        else:
+            logging.info("guest FS is still freeze status")
+        self.gagent.verify_fsfreeze_status(self.gagent.FSFREEZE_STATUS_FROZEN)
+
+        if vm.monitor.cmd("system_reset"):
+            self.gagent.verify_fsfreeze_status(self.gagent.FSFREEZE_STATUS_THAWED)
```

Comment:
Do you need an `else` here to fail the case if above command does not perform as expected?